### PR TITLE
Tmi2 573 

### PR DESCRIPTION
**Pull request by dylanwrightCO**

### CODE REVIEW

File: src/main/java/gov/cabinetoffice/gapfindapiadmin/services/ApiGatewayService.java [hunk 1]
```diff
@@ -119,8 +125,9 @@ private String hashString(String hashingValue) throws NoSuchAlgorithmException {
     public void deleteApiKey(GapApiKey apiKey, boolean isSuperAdmin) {
         log.info("deleting Api key from api gateway with id: {}", apiKey.getApiGatewayId());
 
-        final GrantAdmin grantAdmin = (GrantAdmin) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
-        ofNullable(grantAdmin).filter(admin -> admin.getFunder().getName().equals(apiKey.getFundingOrganisation().getName()) || isSuperAdmin)
+        final JwtPayload jwtPayload = (JwtPayload) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        ofNullable(jwtPayload).filter(admin -> jwtPayload.getDepartmentName()
```
Comment (dominicwest) [inline src/main/java/gov/cabinetoffice/gapfindapiadmin/services/ApiGatewayService.java:129]: Mention of admin here

File: src/main/java/gov/cabinetoffice/gapfindapiadmin/services/TechSupportUserService.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class TechSupportUserService {
+
+    private final TechSupportUserRepository techSupportUserRepository;
+
+    public TechSupportUser getTechSupportUserBySub(String userSub) {
+        return techSupportUserRepository.findByUserSub(userSub);
```
Comment (dominicwest) [inline src/main/java/gov/cabinetoffice/gapfindapiadmin/services/TechSupportUserService.java:17]: Should we use an optional here to be robust?

File: src/main/java/gov/cabinetoffice/gapfindapiadmin/services/ApiGatewayService.java (hunk 1 above)
Comment (GavCookCO) [reply src/main/java/gov/cabinetoffice/gapfindapiadmin/services/ApiGatewayService.java:129]: you're a mention of admin

Comment (GavCookCO) [reply src/main/java/gov/cabinetoffice/gapfindapiadmin/services/ApiGatewayService.java:129]: fixed x

**Review by GavCookCO — APPROVED** [review]

**Comment by GavCookCO** [comment]
Need to merge this into develop and then release branch

**Review by GavCookCO — APPROVED** [review]

**Review by dominicwest — APPROVED** [review]

**Pull request merged**